packages/app-mobile/root.tsx

```diff
@@ -887,6 +903,7 @@ class AppComponent extends React.Component {
 							value: percent,
 						});
 					}}
+					key={this.state.sideMenuWidth}
```

Comment:
That can't be right, or this is a hack. Please explain

Reply:
The side-menu component wasn't being updated (rerendered) to use the new `sideMenuWidth` state whenever the device width changes( ie. when the device orientation changes). So I assigned the `sideMenuWidth` state as a key to the side menu component to cause a rerender each time the `sideMenuWidth` state changes.

Reply:
That doesn't make sense because if I look at the SideMenu component it's not using sideMenuWidth anywhere. If it does need this property you need to pass it to the component and it needs to use it.

You have an onLayoutChange in SideMenu, maybe you need that to set the height, but for the width it should come from sideMenuWidth.

If you do that it will rerender without having to use hacks and hidden dependencies.